container/rules/debian_pkg_tar.bzl

```diff
@@ -0,0 +1,185 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_docker//container:container.bzl", _container_imge = "container")
+load("@base_images_docker//package_managers:download_pkgs.bzl", _download_deb_pkgs = "download")
+load("@base_images_docker//package_managers:apt_key.bzl", _apt_key = "key")
+
+debian_pkgs_attrs = _container_imge.image.attrs + _apt_key.attrs + _download_deb_pkgs.attrs + {
+    # Redeclare following attributes as non-mandatory.
+    "image_tar": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "image": attr.label(
+        allow_files = True,
+        single_file = True,
+    ),
+    "packages": attr.string_list(),
+    "keys": attr.label_list(
+        allow_files = True,
+    ),
+}
+
+InstallableTarInfo = provider(fields = [
+    "installables_tar",
+])
+
+def _generate_deb_tar(
+        ctx,
+        packages = None,
+        additional_repos = None,
+        keys = None,
+        download_pkgs_output_tar = None,
+        download_pkgs_output_script = None):
+    """A function for producing a tarball.
+
+    _generate_deb_tar function produces a tarball with all debian packages, additional_repos, and keys.
```

Comment:
The tar does not have the additonal repos or keys, the additional repos or keys are necessary in order to get the packages.

Reply:
Thanks for the clarification.